### PR TITLE
Add synopsys include path generation

### PR DESCRIPTION
**Pull request by uge**
Creates a target to create a synthesis friendly variable of include paths

### CODE REVIEW

**Comment by bluewww** [comment]
Thanks for your contribution. Can you briefly show the type of file that is generated from this and how you use it in your flow?

Otherwise looks good to me. Since its orthogonal to the other code I don't mind merging it.

**Comment by uge** [comment]
Use case is to create a .tcl file to construct a aggregate list of include paths which end up being fed to the Genus init_hdl_search_path setting for resolving `includes.

Thank you.

**Comment by bluewww** [comment]
If you say it is for genus then you should name the functions appropriately.
edit: I see this can also be used with other tools

File: ipstools/synopsys_defines.py [hunk 1]
```diff
@@ -25,3 +25,6 @@
 SYNOPSYS_ADD_IPS_FILES_CMD = "source scripts/analyze/ips/%s\n"
 SYNOPSYS_ADD_RTL_FILES_CMD = "source scripts/analyze/rtl/%s\n"
 
+SYNOPSYS_INC_DIRS_PREAMBLE  = "set synthesis_include_paths [ list \"%s/%s/include\" \\\n"
```
Comment (bluewww) [inline ipstools/synopsys_defines.py:28]: `include` should be `includes`. If you check the `rtl` subdirectory, you see the folder is called `includes`

**Comment by uge** [comment]
We use genus in our flow but I assume that it would be more usefully named
as synopsys is more common for front end synthesis.

**Comment by bluewww** [comment]
Thanks its merged

**Pull request closed without merging**